Source/RunActivity/Viewer3D/MSTSSky.cs

```diff
@@ -652,58 +644,50 @@ public override void Render(GraphicsDevice graphicsDevice, IEnumerable<RenderIte
             MSTSSkyShader.WindSpeed = Viewer.World.MSTSSky.mstsskywindSpeed;
             MSTSSkyShader.WindDirection = Viewer.World.MSTSSky.mstsskywindDirection; // Keep setting this after Time and Windspeed. Calculating displacement here.
 
+            for (var i = 0; i < 5; i++)
+                graphicsDevice.SamplerStates[i] = SamplerState.LinearWrap;
```

Comment:
How do we know there will always be 5 sampler states to set? (Not a graphics guy.)

Reply:
There are 5 sampler_states defined in the SkyShader.fx effect file, this is where 5 comes from.

Reply:
Got it, thanks. Maybe consider annotating it `SKYSHADER_SAMPLER_STATES` for mortals like me?

Reply:
It's a fair request. :-) In long term however a single sampler object would be enough, given that all of them are the same, but as of now the shaders syntax is kept in the now-deprecated "effect" style, where the sampler object is yet bound to the texture.